flink-runtime/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -220,24 +197,15 @@ private StreamingJobGraphGenerator(
         this.defaultStreamGraphHasher = new StreamGraphHasherV2();
         this.legacyStreamGraphHashers = Arrays.asList(new StreamGraphUserHashHasher());
 
-        this.jobVertices = new LinkedHashMap<>();
         this.builtVertices = new HashSet<>();
-        this.chainedConfigs = new HashMap<>();
-        this.vertexConfigs = new HashMap<>();
-        this.chainedNames = new HashMap<>();
-        this.chainedMinResources = new HashMap<>();
-        this.chainedPreferredResources = new HashMap<>();
-        this.chainedInputOutputFormats = new HashMap<>();
-        this.physicalEdgesInOrder = new ArrayList<>();
         this.serializationExecutor = Preconditions.checkNotNull(serializationExecutor);
-        this.chainInfos = new HashMap<>();
-        this.opNonChainableOutputsCache = new LinkedHashMap<>();
-
+        this.jobVertexBuildContext =
+                new JobVertexBuildContext(streamGraph, new AtomicBoolean(false));
```

Comment:
It is not necessary in this part. The reason for passing this value in the constructor is only because it needs to be explicitly provided in the future AdaptiveGraphManager.

Reply:
> Is it a must to pass in the default value for `hasHybridResultPartition` through the ctor?

It is not necessary in this part. The reason for passing this value in the constructor is only because it needs to be explicitly provided in the future AdaptiveGraphManager.